ReactiveCocoa/Swift/Signal.swift

```diff
@@ -571,6 +571,45 @@ public func takeWhile<T, E>(predicate: T -> Bool)(signal: Signal<T, E>) -> Signa
 	}
 }
 
+/// Zips elements of two signals into pairs. The elements of any Nth pair
+/// are the Nth elements of the two input signals.
+public func zipWith<T, U, E>(otherSignal: Signal<U, E>)(signal: Signal<T, E>) -> Signal<(T, U), E> {
+	return Signal { observer in
+		let lock = NSRecursiveLock()
```

Comment:
I dunno. Hopefully the compiler can optimize them structs. Guess we'd have to inspect IL or assembly. Either way, premature optimization, no?